### ISSUE TITLE
Can texi files be exported? 

### ISSUE THREAD

**Issue by matsl** [melpa/melpa#8989]
It noticed that although being part of the recipe a texi files is not exported to the package tar file. I understand that the texi file is processed and the resulting info file is exported with a dir file. 

With hyperbole we would also want the texi file to be exported. The rational for us is to keep what files get installed the same independent of from which repository hyperbole is being installed. Is there any way to write the recipe so that the texi file is exported?

**Comment by purcell** [comment]
Hmm, I don't think that's possible currently. During the [build process](https://github.com/melpa/package-build/blob/236ef4e4d615699d4ba8b7ad28bd40b87d269411/package-build.el#L986-L1013) we actually remove the old texi file from the working directory, in order to keep only the generated `.info` file. In general we encourage people to omit from the package any documentation that's not in Info format, e.g. README.md or even README.txt. Does the absence of that file cause a functional issue for hyperbole?

**Comment by matsl** [comment]
> During the [build process](https://github.com/melpa/package-build/blob/236ef4e4d615699d4ba8b7ad28bd40b87d269411/package-build.el#L986-L1013) we actually remove the old texi file from the working directory,

Yes, that was what it looked like. I just wanted to double check if I was missing something and there would be a way around that. Thanks for confirming.

>  Does the absence of that file cause a functional issue for hyperbole?

No, there is no functional issue. It is only for the user to be able to view and alter the source of the documentation without doing any further download.

**Comment by purcell** [comment]
Thanks, I understand. I think overall this isn't something we'd be likely to change, sorry.

**Comment by rswgnu** [comment]
Mats is a little off here.  hyperbole.texi is part of the source of Hyperbole and therefore should be part of the package installed.  There are links in other files to this file, so if it is not there, those links won't work.  We are already including all of the formatted versions of the manual, so including this file will add little extra space to the installed package.  The recipe says to include it already; can you do something just for Hyperbole to allow this?

**Comment by tarsius** [comment]
I would be willing to implement support for allowing only certain packages to use the bubblewrapped org->texi export. But it is not up to me @purcell and @milkypostman would have to decide whether to allow this.

If we do this, I would like to bless a few of my own packages.

But what would we do if someone less well-known whose motives we don't know asks for their packages to be blessed?

**Comment by tarsius** [comment]
I misunderstood. I thought this was about generating `hyperbole.texi` from something else, likely `hyperbole.org`. That is supported by `package-build` but not enabled on MELPA for security reasons.

But you are asking to *include* (the verb "export" confused me) `hyperbole.texi` in the tarball, in addition to the `hyperbole.info` that we generate from it. As @purcell said, we don't support that and we are unlikely to add support. I do not want to implement for this in `package-build.el`. As far as I know, Hyperbole is the only package that needs that, and I don't want to add features for a single package.

You mention that other files link to `hyperbole.texi`. If at all possible / reasonable, I recommend you change those links to point at `hyperbole.info` instead. Maybe consider this a feature suggestion for Hyperbole itself: implement a link type that points at a info file, but if that is absent, falls back to the respective texi file (or vice-versa, I don't know what makes more sense for your usecase).

**Comment by minad** [comment]
@tarsius Is there interest in supporting org->texi compilation? My packages don't provide texi files, such that the MELPA builds lack info files, in contrast to the ELPA builds.

**Comment by tarsius** [comment]
I have implemented org->texi compilation in `package-build` a long time ago, but @purcell does not want to enable that for security reasons.

**Comment by rswgnu** [comment]
The issue Mats brought up is providing the tarball of a package as one type of source distribution not a run-time only distribution.  If it includes the .el source files, then it should include the .texi source form of the package manual.

Is not the point of the tarball to be a source distribution?

**Comment by minad** [comment]
@tarsius 

> I have implemented org->texi compilation in package-build a long time ago, but @purcell does not want to enable that for security reasons.

Yes, understood. I wondered if the security reasons are really not solvable (you mentioned bubblewrap) or if it is more about avoiding the additional complexity? As far as I understand, ELPA builds every package in a sandbox.

**Comment by tarsius** [comment]
I think "it should be safe enough" (famous last words). But I am not really qualified to make that determination, and respect @purcell's decision.
